docs/testing-guidelines/getting-code-coverage.md

```diff
@@ -0,0 +1,127 @@
+Getting Code Coverage Analysis for PowerShell
+===
+
+**Note: Code coverage is currently only supported on Windows, since we use OpenCover**
+
+The PowerShell code base is configured to build with code coverage support using [OpenCover](https://github.com/OpenCover/opencover). You can see the testing coverage of the current [`master`](https://github.com/PowerShell/PowerShell) branch build at any time at [codecov.io](https://codecov.io/gh/PowerShell/PowerShell).
+
+To run test coverage analysis of PowerShell on your own branch/machine
+however, you will need to take the following steps (and be aware that running
+the tests under code coverage analysis can take as long as 8 hours).
+
+Running tests with code coverage analysis
+---
+
+**First**: Open PowerShell in an **Administrator** session. OpenCover
+needs Administrator privileges to work.
+
+Now, in PowerShell:
+```powershell
+# Go to your PowerShell build directory root
+PS> Set-Location "C:\Path\to\powershell\build\dir"
+
+# Import the PowerShell build module
+PS> Import-Module .\build.psm1
+
+# Build PowerShell. You may need to add other flags here like
+# -ResGen, -Restore or -PSModuleRestore
+PS> Start-PSBuild -Configuration CodeCoverage
+
+# Save the built executable's location for later use
+PS> $psDir = Split-Path -Parent (Get-PSOutput)
+
+# Now ensure Pester is installed
+PS> Restore-PSPester
+
+# Import the OpenCover module
+PS> Import-Module $PWD\test\tools\OpenCover
+
+# Install OpenCover to a temporary directory
+PS> Install-OpenCover -TargetDirectory $env:TEMP -Force
+
+# Finally, run the tests with code coverage analysis.
+# If you want to run only the continuous integration tests,
+# add -CIOnly, which will take less time
+PS> Invoke-OpenCover `
+>> -OutputLog coverage.xml `
+>> -TestPath $PWD\test\powershell `
+>> -OpenCoverPath $env:TEMP\OpenCover `
+>> -PowerShellExeDirectory $psDir
+```
+
+Examining the code coverage data
+---
+
+Once the code coverage test run is done, you'll want to examine the data:
+
+```powershell
+# Collect the coverage data using Get-CodeCoverage from the OpenCover
+# module that was imported above
+PS> $coverageData = Get-CodeCoverage .\coverage.xml
```

Comment:
May be worth mentioning that this is an expensive operation so recommendation is to store it in a variable for reuse/re-viewing